### ISSUE TITLE
[labels] Removing, updating labels on issues can be simpler

### ISSUE THREAD

**Issue by karlcow** [webcompat/webcompat.com#797]
When requesting an issue and updating the labels. This is what we do currently 
```
127.0.0.1 - - [23/Oct/2015 07:00:20] "GET /issues/284 HTTP/1.1" 200 -
127.0.0.1 - - [23/Oct/2015 07:00:20] "GET /api/issues/labels HTTP/1.1" 200 -
127.0.0.1 - - [23/Oct/2015 07:00:30] "POST /api/issues/284/labels HTTP/1.1" 200 -
127.0.0.1 - - [23/Oct/2015 07:00:41] "GET /api/issues/284 HTTP/1.1" 200 -
```

The content being sent with the POST is:

```python
('[{"url":"https://api.github.com/repos/webcompat/webcompat-tests/labels/closed-invalid","name":"closed-invalid","color":"e6e6e6"},
{"url":"https://api.github.com/repos/webcompat/webcompat-tests/labels/status-bug","name":"status-bug","color":"fc2929"},
{"url":"https://api.github.com/repos/webcompat/webcompat-tests/labels/status-imported","name":"status-imported","color":"bfd4f2"}]', 
200, 
{'etag': 'W/"8ac00d04b6fe55da683d748efc5ed980"', 'content-type': 'application/json', 'cache-control': 'private, max-age=60, s-maxage=60'})
```

I was checking the [labels API for issues in the documentation](https://developer.github.com/v3/issues/labels/#add-labels-to-an-issue) and I see that the current version of the API is a lot simpler.  We can just `POST /repos/:owner/:repo/issues/:number/labels` with 

```
[
  "Label1",
  "Label2"
]
```

And if we want to replace all labels by another set probably our case

`PUT /repos/:owner/:repo/issues/:number/labels` with 

```
[ 
  "Label1",
  "Label2"
]
```

**Comment by miketaylr** [comment]
I nominate @hallvors to provide feedback on this once https://github.com/webcompat/webcompat.com/pull/796 lands.

**Comment by karlcow** [comment]
As a reminder, we want probably to not send an empty `PUT` that will delete all labels, except if we really want to. Meaning: we need to make extra care and separate the case where the GET failed from the user changing labels.

**Comment by hallvors** [comment]
#796 will fix this. :)
I don't have a good solution to the GET failed case ATM (but dropping the cache will solve most of it - hardcoding labels if we go down that route, which I support and consider a simple change, will also help. In the new code, we could disable label editing completely if ```issues.allLabels.toArray().length < 1``` but we're basically taking care of this problem by making it not happen rather than just treating the symptoms ;)).

**Comment by hallvors** [comment]
New code causes these requests:
```
127.0.0.1 - - [24/Oct/2015 12:44:30] "GET /issues/2 HTTP/1.1" 200 -
127.0.0.1 - - [24/Oct/2015 12:44:33] "GET /api/issues/2 HTTP/1.1" 200 -
127.0.0.1 - - [24/Oct/2015 12:44:58] "POST /api/issues/2/labels HTTP/1.1" 200 -
```
POST content is for example:
```
["browser-firefox","os-android","browser-chrome","closed-fixed"]
```

**Comment by hallvors** [comment]
This was fixed by merging #796
